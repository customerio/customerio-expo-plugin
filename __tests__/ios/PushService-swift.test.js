const fs = require("fs-extra");
const path = require("path");

const testProjectPath = path.join(__dirname, "../../test-app");
const iosPath = path.join(testProjectPath, "ios");
const pushServicePath = path.join(iosPath, "ExpoTestApp/PushService.swift");

test("Plugin creates expected PushService.swift", async () => {
  const content = await fs.readFile(pushServicePath, "utf8");

  expect(content).toMatchSnapshot();
});
