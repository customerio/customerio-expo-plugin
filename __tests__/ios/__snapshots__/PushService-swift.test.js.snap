// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin creates expected PushService.swift 1`] = `
"import Foundation
import CioMessagingPushAPN
import UserNotifications
import UIKit

@objc
public class CIOAppPushNotificationsHandler : NSObject {

  public override init() {}

  

  @objc(initializeCioSdk)
  public func initializeCioSdk() {
    MessagingPushAPN.initialize(
      withConfig: MessagingPushConfigBuilder()
        .showPushAppInForeground(true)
        .autoTrackPushEvents(true)
        .build()
    )
  }

  @objc(application:deviceToken:)
  public func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
    MessagingPush.shared.application(application, didRegisterForRemoteNotificationsWithDeviceToken: deviceToken)
  }

  @objc(application:error:)
  public func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
    MessagingPush.shared.application(application, didFailToRegisterForRemoteNotificationsWithError: error)
  }
}
"
`;
